apps/server/src/modules/school/api/mapper/years.response.mapper.ts

```diff
@@ -0,0 +1,66 @@
+import { School, SchoolYear } from '../../domain';
+import { SchoolYearResponse, YearsResponse } from '../dto/response';
+import { MissingYearsLoggableException } from '../error/missing-years.loggable-exception';
+
+export class YearsResponseMapper {
+	public static mapToResponse(school: School, schoolYears: SchoolYear[]): YearsResponse {
+		const schoolYearResponses = schoolYears.map((schoolYear) => schoolYear.getProps());
+		const activeYear = this.computeActiveYear(school, schoolYears);
+		const nextYear = this.computeNextYear(schoolYears, activeYear);
+		const lastYear = this.computeLastYear(schoolYears, activeYear);
+
+		const res = {
+			schoolYears: schoolYearResponses,
+			activeYear,
+			lastYear,
+			nextYear,
+		};
+
+		return res;
+	}
+
+	private static computeActiveYear(school: School, schoolYears: SchoolYear[]): SchoolYearResponse {
```

Comment:
Es wäre auch möglich die privaten Methoden alle an das Do zu hängen. Ich würde erwarten das sie in Teilen auch bei Sync Operationen notwendig sind.

Reply:
Die Methoden könnte man nur ans DO hängen, wenn man auch alle SchoolYears aus der Datenbank ans DO hängt. Da bin ich aber eher dagegen, weil die SchoolYears nicht wirklich zu einer einzelnen Schule gehören.

Reply:
Hab jetzt so ne Utils-Klasse erstellt.